### PR TITLE
[#11732] Broken PUML diagrams in developer guide

### LINKED ISSUE #11732: Broken PUML diagrams in developer guide
Raised in https://github.com/TEAMMATES/teammates/pull/11680#issuecomment-1098418311
- **Environment**: 

**Steps to reproduce**
Current developer guide is displaying this behaviour.

However, I can't seem to reproduce this on my own branch. The issue seemed to be there but once I built and deployed it showed no error: See https://kaixin-hc.github.io/teammates/design.html which was built from the master branch which should be currently synced with the teammates master. I've asked the MarkBind team to investigate with me. 

**Expected behaviour**

The PUML diagram should be displayed.

**Actual behaviour**
The following error seems to be displayed in most cases.
Screenshot 2022-04-14 at 12 50 43 PM

**Additional info**

As the devGuide is actively used by developers, I will push a temporary change to use .png files until we can figure out why this issue is occurring.

### PR DESCRIPTION
**Pull request by tlylt**
Fixes #11732
Supersedes and hence closes ##11733

**Outline of Solution**
PlantUML diagrams are used in dev docs (Design), which MarkBind requires both Java and Graphviz to be installed, in order to generate these diagrams on build.

Let's add an installation step for Graphviz (Java is available by default since we use `ubuntu-lastest`) in CI such that the diagrams can be generated before deploying to gh-pages.
Let's also include such details in our dev docs (Documentation), to remind developers who intend to contribute to teammates' documentation to install the required dependencies.

The expected outcome of the dev docs page on "design" (that contains broken images) is available at this [link](https://tlylt.github.io/test-docs/design.html) for inspection.

**Others**
There are also some additional issues reported/found such as
- readme does not refer to the deployed dev docs
- pull request template does not refer to the deployed dev docs
- The issue bot does not refer to the deployed dev docs

Will make separate issues for these.

### CODE REVIEW

File: docs/documentation.md [hunk 1]
```diff
@@ -13,6 +13,11 @@ All the commands in this document are assumed to be run from the `/docs` folder,
 1. Install `Node.js` (minimum version 12).
 1. Run `npm ci` to install the necessary tools to build documentation, including MarkBind.
 
+Install the following additional dependencies required by MarkBind to generate [PlantUML](https://plantuml.com/) diagrams locally:
+
+1. Install [Java](https://www.java.com/en/download/) 8 (or later).
```
Comment (wkurniawan07) [inline docs/documentation.md:18]: No need to add link here. Java is a general tool, and we don't want to give the wrong impression of being constrained to use Oracle JDK. (e.g. I myself use Amazon Corretto)
```suggestion
1. Install Java 8 or later.
```

File: .github/workflows/dev-docs.yml [hunk 2]
```diff
@@ -21,6 +21,8 @@ jobs:
       - uses: actions/setup-node@v2
         with:
           node-version: 12
+      - name: Install Graphviz
```
Comment (wkurniawan07) [inline .github/workflows/dev-docs.yml:24]: Not related to this line, but do we need to have `actions/setup-java@v1`, at least to specify the version?

**Review by wkurniawan07 — CHANGES_REQUESTED** [review]
Thanks for the quick turnaround

File: .github/workflows/dev-docs.yml (hunk 2 above)
Comment (tlylt) [reply .github/workflows/dev-docs.yml:28]: It should work without explicit installation as java 11 is available in ubuntu-latest, but adding it to specify the version is fine as well. I have updated it accordingly.

Comment (wkurniawan07) [reply .github/workflows/dev-docs.yml:28]: > It should work without explicit installation as java 11 is available in ubuntu-latest

Making this assumption means we're relying on internal settings of the machine, which may change in a way that can break the build.

**Review by wkurniawan07 — APPROVED** [review]

**Pull request merged**